### PR TITLE
Update: Collisions reworked For Explosions

### PR DESCRIPTION
**Pull request by Migu3lito5**
In this PR:
- New Layer -> "Collisions"
- Most prefabs given this new layer tag
- Explosion.cs

In Explosion.cs, I changed the way collisions are detected to a OverlapSphere, with layers involved. This made the performance better with less unwanted collisions.

### CODE REVIEW

File: Seasons Defense/Assets/Scripts/Explosion.cs [hunk 1]
```diff
@@ -22,43 +22,40 @@ private void Start()
...
+            hit.GetComponent<Missile>().BeingDestroyed();
         }
-        
+
         // if a tower -> disable it 
-        else if (obj.GetComponent<Tower>()) 
+        else if (hit.GetComponent<Tower>())
         {
-            obj.GetComponent<Tower>().DisableTurret();
+            hit.GetComponent<Tower>().DisableTurret();
```
Comment (ConlynPattison) [inline Seasons Defense/Assets/Scripts/Explosion.cs:48]: Inside of the Turret.DisableTurret function, let's disable the collider of that tower so we don't have 50+ collisions with an already disabled tower.

**Review by ConlynPattison — CHANGES_REQUESTED** [review]

File: Seasons Defense/Assets/Scripts/Explosion.cs (hunk 1 above)
Comment (Migu3lito5) [reply Seasons Defense/Assets/Scripts/Explosion.cs:48]: Check out the change

File: Seasons Defense/Assets/Scripts/Tower.cs [hunk 2]
```diff
@@ -62,15 +62,22 @@ public void DisableTurret()
...
         LevelManager.Instance.BuildingDestroyed();
         
         UpdateAmmoCount();
     }
+
+    public void EnableTower()
+    {
+        // TODO: enables tower for the next wave.
+    }
+
```
Comment (ConlynPattison) [inline Seasons Defense/Assets/Scripts/Tower.cs:80]: Looks good. Thank you!

**Review by ConlynPattison — APPROVED** [review]
Thank you for making the change. This looks good.

**Pull request merged**